src/libappimage/desktop_integration.c

```diff
@@ -526,7 +526,8 @@ bool move_diricon_as_app_icon(const char* tempdir_path, const char* user_data_di
     char* target_dir_path = g_build_path("/", user_data_dir, "icons/hicolor/32x32/apps/", NULL);
     g_mkdir_with_parents(target_dir_path, S_IRWXU);
 
-    char* new_icon_name = g_strjoin("_", vendorprefix, appimage_path_md5, icon_name, NULL);
+    char* icon_name_with_extension = g_strjoin("", icon_name, ".png", NULL);
```

Comment:
Any reason for the leading `""`?

Reply:
It's the separator that will be used to join the two strings. "" mean that no separator will be used. You know glib rocks.